internal/util/render/executor.go

```diff
@@ -405,6 +406,29 @@ func (pn *pkgNode) runMutators(ctx context.Context, hctx *hydrationContext, inpu
 	}
 
 	for i, mutator := range mutators {
+		if pl.Mutators[i].ConfigPath != "" {
+			// kpt v1.0.0-beta15+ onwards, functionConfigs are included in the
+			// function inputs during `render` and as a result, they can be
+			// mutated during the `render`.
+			// So functionConfigs needs be updated in the FunctionRunner instance
+			// before every run.
+			for _, r := range input {
```

Comment:
It strikes me as inefficient to iterate through all the input resources to search for the functionConfig between each run, but admittedly I can't think of a more efficient solution without overcomplicating things.

Reply:
Agreed.